compiler/resolution/visibleFunctions.cpp

```diff
@@ -71,12 +75,31 @@ static int                                    nVisibleFunctions       = 0;
 *                                                                             *
 ************************************** | *************************************/
 
+static void getVisibleMethods(const char* name, CallExpr* call,
+                              Vec<FnSymbol*>& visibleFns);
+
 static void  buildVisibleFunctionMap();
 
 void findVisibleFunctions(CallInfo&       info,
                           Vec<FnSymbol*>& visibleFns) {
   CallExpr* call = info.call;
 
+  if (!builtTypeHelperNames) {
+    // Build the cache of names we care about even though they aren't methods
+    typeHelperNames.insert(astrSassign);
+    typeHelperNames.insert(astrSeq);
+    typeHelperNames.insert(astrSne);
+    typeHelperNames.insert(astrSgt);
+    typeHelperNames.insert(astrSgte);
+    typeHelperNames.insert(astrSlt);
+    typeHelperNames.insert(astrSlte);
+    typeHelperNames.insert(astrSswap); // ?
+    typeHelperNames.insert(astr_cast);
+    typeHelperNames.insert(astr_defaultOf);
+    typeHelperNames.insert(astrNew);
```

Comment:
I would think this would need initCopy / autoCopy / autoDestroy as well. That probably also indicates you'll need to add some tests that make a record with non-trivial `init=` and `deinit` and arrange for the type to be visible but not the methods... and then make sure it works to make a couple of copies. Try returning a `const ref` argument by value - that's a pretty sure way to get a copy...